### PR TITLE
S2 frontend single question view

### PR DESCRIPTION
**Pull request by NunoDasNeves**
Not actually done yet... Will add input for submitting answers

### CODE REVIEW

File: frontend/src/components/AnswerCard.vue [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+</script>
+
+<style scoped>
+.date {
+  font-size: 0.75em;
+  color: rgba(0, 0, 0, 0.65);
+}
+
+.card {
```
Comment (DarkPurple141) [inline frontend/src/components/AnswerCard.vue]: likely to abstract part of this out in next PR.

File: frontend/src/views/Question.vue [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+      <ul>
+          <li v-for="answer in answers" :key="answer.id">
+              <AnswerCard :answer="answer"/>
+          </li>
+      </ul>
     </div>
 </template>
 
 <script>
+import QuestionCard from '@/components/QuestionCard.vue'
```
Comment (DarkPurple141) [inline frontend/src/views/Question.vue]: vue file extension not required in imports. (Webpack is able to surmise)

File: frontend/src/views/Question.vue [hunk 3]
```diff
@@ -26,3 +33,9 @@ export default {
   }
 };
 </script>
+
+<style scoped>
```
Comment (DarkPurple141) [inline frontend/src/views/Question.vue]: probably should put normalised style like this into root.

**Comment by DarkPurple141** [comment]
Just added some stuff to this branch Nunez.

**Comment by NunoDasNeves** [comment]
I still need to write tests I suppose ... will tackle tomorrow

File: frontend/src/views/Question.vue [hunk 4]
```diff
@@ -33,6 +41,37 @@ export default {
...
+  },
+  methods: {
+    submitAnswer : function (answerForm) {
+        // check that they actually typed something
+        if (answerForm.body == "") {
+            this.answerFormResponse.text = "Please type an answer!"
+            this.answerFormResponse.style = {'form-success': false, 'form-failure': true}
+            return
+        }
+        // post the answer to the api
```
Comment (DarkPurple141) [inline frontend/src/views/Question.vue]: This should really be handled in the store.

Comment (DarkPurple141) [reply frontend/src/views/Question.vue]: Also strictly beyond scope of current issue.

File: frontend/src/views/Question.vue [hunk 5]
```diff
@@ -33,6 +41,37 @@ export default {
...
+  data: function () {
+    return {
+      answerFormResponse: {
+        text: "",
+        style: {'form-success': false, 'form-failure': false}
+      }
+    }
+  },
+  methods: {
+    submitAnswer : function (answerForm) {
```
Comment (DarkPurple141) [inline frontend/src/views/Question.vue]: this syntax is a bit old worldy. New syntax for that kind of func is just:

```js
submitAnswer(args) {

}
```

File: frontend/src/router.js [hunk 6]
```diff
@@ -23,10 +23,8 @@ export default new Router({
     {
       path: '/question/:id',
       name: 'question',
-      props: ({params: { id }}) => ({
-        id
-      }),
-      component: () => import(/* webpackChunkName: "about" */ './views/Question.vue')
+      props: ({params: { id }}) => { id },
```
Comment (DarkPurple141) [inline frontend/src/router.js]: This is a bug as it stands.
() => {} // doesn't return an object
() => ({}) returns an empty object

File: frontend/src/utils/api.js [hunk 7]
```diff
@@ -35,6 +36,15 @@ export function postQuestion (body) {
 }
 
 /**
+ * @param String id the id of the question being answered
```
Comment (DarkPurple141) [inline frontend/src/utils/api.js]: JSDoc format is @param {type} name description

File: frontend/src/components/AnswerForm.vue [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+<script>
+import Card from './Card'
+
+export default {
+  name: 'AnswerForm',
+  components: {
+      Card
+  },
+  data: function () {
```
Comment (DarkPurple141) [inline frontend/src/components/AnswerForm.vue]:
```js
data() {
  return {
    ...
   }
}
```

File: frontend/src/router.js (hunk 6 above)
Comment (DarkPurple141) [reply frontend/src/router.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

**Comment by NunoDasNeves** [comment]
hnnnnnnnng

File: frontend/src/router.js (hunk 6 above)
Comment (NunoDasNeves) [reply frontend/src/router.js]: oic

**Pull request merged**